### ISSUE TITLE
Microstutters only present in flathub/flatpak release of PolyMC

### ISSUE THREAD

**Issue by Havard-SL** [flathub/org.polymc.PolyMC#20]
Hi!
I've been looking into performance issues I've been getting on PolyMC during the last evening, and I've encountered a rather weird issue.

It seems like this release of PolyMC has many more, and noticeable microstutters than the non-flatpak release.

I've attatched a couple pictures below. The client on the right is this release, while the client on the left is an rpm release (I run fedora 36)

These pictures are cherrypicked as there are moments where neither has any lagspikes, but the flatpak release has many more microstutters every 3-5seconds, and can last from 70ms-500ms. How long it lasts seem to be random.

![Screenshot from 2022-05-14 21-50-31](https://user-images.githubusercontent.com/55786766/168446559-50744db3-7480-4c91-bf79-73796b01bc67.png)
![Screenshot from 2022-05-14 21-50-45](https://user-images.githubusercontent.com/55786766/168446563-6143119b-cb2f-4c28-834d-5de4f6a35963.png)
![Screenshot from 2022-05-14 21-51-12](https://user-images.githubusercontent.com/55786766/168446583-9af32d84-6236-44fd-a644-40d9bdf3d8af.png)

(The flatpak release on the right is identical, with the exception of some different java parameters, but it is accurate to how it runs with no parameters set)

I encounter this under both wayland and X-Org.

Another thing I encountered during testing was that any lag-spike encountered on the flatpak release would be reflected in wireplumber writing to disk in system monitor. I took a look at wireplumber with some debug flags enabled, and it seems something weird happened while the spikes occured, but I could not locate any specific issues as I have no clue what it does and how it does it :P

If you need any more info, please let me know!

Kind regards,
Håvard

**Comment by dada513** [comment]
Hello, 
thanks for the detailed bug report. Are you using an SSD or HDD drive?

**Comment by Havard-SL** [comment]
Hi!
I have my root on an SSD, and my home on an HDD. Both on different btrfs subvolumes.

**Comment by dada513** [comment]
Can you test if the same issue appears on the official minecraft launcher flatpak?

**Comment by Havard-SL** [comment]
Why didn't I test that earlier!

It seems the same issues appear in the flathub release of the minecraft launcher. 
However, like with polymc, it does not happen on the AppImage release of the minecraft launcher.

Is this therefore a flatpak issue?

**Comment by dada513** [comment]
yes it probably is, I am going to try to debug this maybe submit to upstream flatpak  
can you check your mesa version on your host? are you using nvidia graphics?

**Comment by Havard-SL** [comment]
I'm sorry if I misunderstood your question. I don't think I use mesa on my computer, but the NVIDIA proptietary drivers version 510.68.02.
